ui/app/models/kv/metadata.js

```diff
@@ -77,6 +77,21 @@ export default class KvSecretMetadataModel extends Model {
     return array.reverse();
   }
 
+  // helps in long logic statements for state of a currentVersion
+  get currentSecret() {
+    const currentVersionObject = this.versions[this.currentVersion];
```

Comment:
just my opinion 💭  long variable names can bonk me up sometimes. since `this.currentVersion` is passed in i feel like this being the current version data is intuitive and perhaps this could be renamed to just `data`